### PR TITLE
DDF-1319 - continued clean up and refactoring to have JAXRS sources use the SecureCxfClientFactory

### PR DESCRIPTION
**Pull request by kcwire**
@jlcsmith is hero
@bdeining 
@jrnorth 
@wmcnalli 
@beyelerb 
@millerw8 
@shane-voisard 
@EIrwin
@kguerriero

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/150)

### CODE REVIEW

File: catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java [hunk 1]
```diff
@@ -840,42 +763,27 @@ private OpenSearchSource givenSource(Answer<BinaryContent> answer)
...
-
-        Client proxy = mock(Client.class);
-
-        when(openSearchConnection
-                .newRestClient(any(String.class), any(Query.class), any(String.class),
-                        any(Boolean.class))).thenReturn(proxy);
-
-        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);
-
+        WebClient client = mock(WebClient.class);
```
Comment (roelens8) [inline catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java]: This is duplicated from two lines above it?

**Comment by roelens8** [comment]
<sup>**[catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java, line 862 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyEx7GiU0roP82gI4Wv)** ([raw file](https://github.com/codice/ddf/blob/97c37fbecd54bced0b929b83b16825fe46c174e5/catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java#L862)):</sup>
This line requires SecurityServiceException to be handled.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by beyelerb** [comment]
Reviewed 28 of 35 files at r1, 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by kcwire** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java, line 766 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyDDahlBgleXYZayu3x-r2-766)** ([raw file](https://github.com/codice/ddf/blob/97c37fbecd54bced0b929b83b16825fe46c174e5/catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java#L766)):</sup>
certainly is. removed.

---

<sup>**[catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java, line 862 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyEx7GiU0roP82gI4Wv)** ([raw file](https://github.com/codice/ddf/blob/97c37fbecd54bced0b929b83b16825fe46c174e5/catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java#L862)):</sup>
good catch. pushed this up prematurely. Doing a fresh build now and will push up the fixes shortly.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by roelens8** [comment]
Reviewed 22 of 35 files at r1, 4 of 7 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by jrnorth** [comment]
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 795 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIpF3xONaI_lROd454)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L795)):</sup>
Missing a parameter in the Javadoc.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 824 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIpJhUT_QzexE0Pio3)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L824)):</sup>
Missing a parameter in the Javadoc.

---

<sup>**[catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java, line 270 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIsDoXgwHHkBS5URZw)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java#L270)):</sup>
Looks like the Javadoc's out of date.

---

<sup>**[catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java, line 745 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIvim1gC5EulTh2TdL)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java#L745)):</sup>
Did you mean to create an anonymous class? And why 10?

---

<sup>**[catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java, line 314 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIxLNzgoQGSXwNKT0m)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java#L314)):</sup>
"it is missing"

---

<sup>**[catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java, line 393 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIysSMkcQBYqT9mK56)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java#L393)):</sup>
Wfs wfs = getClient()?

---

<sup>**[catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/TestWfsSource.java, line 568 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJ-XB51UDb36egbS_j)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/TestWfsSource.java#L568)):</sup>
Could this exception go on its own line?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by coyotesqrl** [comment]
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

<sup>**[catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java, line 817 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJGlrsrOZJo-d8f9tB)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java#L817)):</sup>
This is really petty, but in all the places we're constructing a `SecureCxfClientFactory`, could we do this...
```
new SecureCxfClientFactory<>(...);
```
That diamond operator is shorthand for the specific type that's being passed in and will quiet tools down a bit when they check for type safety.

---

<sup>**[catalog/pom.xml, line 126 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJI_VOsQp1wm-ptiIq)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/pom.xml#L126)):</sup>
s/b `project.version`

---

<sup>**[catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java, line 1573 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJNpXkKMhNGCc4o2NV)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java#L1573)):</sup>
The order here feels weird to me. It seems `Subject` should take precedence, unless we want to be able to eject subject in favor of new credentials?

---

<sup>**[catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java, line 745 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyIvim1gC5EulTh2TdL)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java#L745)):</sup>
+1 (Or, you know, +6 if you want to get to the `HashMap` default init size).

---

<sup>**[catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java, line 446 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJQ6tukSrYCFezd0vx)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java#L446)):</sup>
```
Wfs wfs = getClient();
```

---

<sup>**[catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java, line 571 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJX1TWPHEaM2VpVzcu)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_0_0/catalog/source/WfsSource.java#L571)):</sup>
This, and then not throwing an exception below, can cause an NPE down on line 632.

---

<sup>**[catalog/spatial/wfs/2.0.0/spatial-wfs-v2_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v2_0_0/catalog/source/WfsSource.java, line 386 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJXgxJQu8l3hBozXqA)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/2.0.0/spatial-wfs-v2_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v2_0_0/catalog/source/WfsSource.java#L386)):</sup>
Init in one call.

---

<sup>**[catalog/spatial/wfs/2.0.0/spatial-wfs-v2_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v2_0_0/catalog/source/WfsSource.java, line 440 \[r3\]](https://reviewable.io:443/reviews/codice/ddf/150#-JyJXmBfWbhQrJv-Yt5x)** ([raw file](https://github.com/codice/ddf/blob/318389e6414e2c01aa2ddfeb02574a6a0aeea760/catalog/spatial/wfs/2.0.0/spatial-wfs-v2_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v2_0_0/catalog/source/WfsSource.java#L440)):</sup>
This can result in an NPE below...

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by jlcsmith** [comment]
A clean build was successful minor a code coverage violation in csw common that I believe is unrelated to this PR.

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by jlcsmith** [comment]
@kcwire - Smoke test results:
* CSW  Source: federation (loop back and external) including basic auth looks good  
* WFS 2.0 Source: it doesn't appear that the authorization header is getting set when connecting using basic auth.  

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/150)

**Comment by pklinef** [comment]
You should be able to squash the "resolving merge issues" commit.

**Comment by coyotesqrl** [comment]
lgtm!

**Pull request merged**